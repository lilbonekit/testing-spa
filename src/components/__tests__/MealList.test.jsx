import { screen } from '@testing-library/react';
import { renderWithRouter } from '../../utils/testing';

import { MealList } from '../MealList';

describe('MealList', () => {
  it('renders correctly', () => {
    renderWithRouter(
      <MealList
        meals={[
          {
            strMeal: 'Cheese cake',
            idMeal: '123',
            strMealThumb: '/meal.png',
          },
          {
            strMeal: 'Chocolate cake',
            idMeal: '456',
            strMealThumb: '/choco.png',
          },
        ]}
      />,
    );

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});
