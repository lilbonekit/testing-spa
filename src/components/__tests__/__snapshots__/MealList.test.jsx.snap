// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealList renders correctly 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Cheese cake"
        src="/meal.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Cheese cake
      </span>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/meal/123"
      >
        Watch recipe
      </a>
    </div>
  </div>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Chocolate cake"
        src="/choco.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Chocolate cake
      </span>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/meal/456"
      >
        Watch recipe
      </a>
    </div>
  </div>
</div>
`;
